### PR TITLE
SNS: added api for ListTopics

### PR DESCRIPTION
**Pull request by alexturkin**
This PR adds "ListTopics" action for SNS service. 
Also a bunch of tests for SNS module are added.

### CODE REVIEW

File: src/erlcloud_sns.erl [hunk 1]
```diff
@@ -543,6 +580,15 @@ sns_request(Config, Action, Params) ->
             erlang:error({sns_error, Reason})
     end.
 
+list_all(Fun, Type, Config, Token, Acc) ->
+    Res = Fun(Token, Config),
+    List = proplists:get_value(Type, Res),
+    case proplists:get_value(next_token, Res) of
+        "" ->
+            lists:append(Acc, List);
```
Comment (ransomr) [inline src/erlcloud_sns.erl]: Calling `lists:append(Acc, List)` will need to copy `Acc` each call. More efficient to use `List ++ Acc` and then reverse the result when you are done.

**Comment by ransomr** [comment]
I'm seeing a couple of new warnings from the tests:
```
test/erlcloud_sns_tests.erl:85: Warning: variable 'Key' exported from 'case' (line 78)
test/erlcloud_sns_tests.erl:89: Warning: variable 'Value' exported from 'case' (line 78)
```

File: src/erlcloud_sns.erl (hunk 1 above)
Comment (smilerlee) [inline src/erlcloud_sns.erl]: Way like `erlcloud_ddb_util:flatreverse/1` is much more efficient

**Comment by alexturkin** [comment]
@ransomr @smilerlee addressed

I've also moved flatreverse/1 to common module erlcloud_util, since it can be used from a lot of places which implement paging queries

**Comment by smilerlee** [comment]
@alexturkin I would prefer not to extract this as a utility function since it's just a tiny implementation trick.

Any way, if I were you to implement this, I would declare an opaque type (for `Acc`), and three functions like:
* `new() -> []`
* `append(Acc, List) -> [List | Acc]`
* `final(Acc)` or `reverse(Acc)`

Perhaps they should reside in a standalone module like `list_appender` or something similar, maybe `erlcloud_` prefix is also needed.

However, I'd rather not to add this little convenience at the expense of complexity and exposing implementation details.

**Comment by alexturkin** [comment]
@smilerlee I think you right, being an utility function is too much. I'll redo this

**Comment by alexturkin** [comment]
@smilerlee please, take a look - I've put this trick just in `erlcloud_sns:list_all` function.

**Comment by smilerlee** [comment]
@alexturkin seems great. I've never seen the usage of `fun erlang:'++'/2` before, it awesome!

**Comment by ransomr** [comment]
Thanks for the contribution.

**Comment by alexturkin** [comment]
@smilerlee @ransomr thank you guys!

**Pull request merged**